src/slice-chop.c

```diff
@@ -312,19 +312,25 @@ static SEXP chop_shaped(SEXP x, SEXP indices, struct vctrs_chop_info info) {
 static SEXP chop_fallback(SEXP x, SEXP indices, struct vctrs_chop_info info) {
   SEXP elt;
 
-  // Construct call with symbols, not values, for performance
-  SEXP call = PROTECT(Rf_lang3(syms_bracket, syms_x, syms_i));
-
   // Evaluate in a child of the global environment to allow dispatch
   // to custom functions. We define `[` to point to its base
   // definition to ensure consistent look-up. This is the same logic
   // as in `vctrs_dispatch_n()`, reimplemented here to allow repeated
   // evaluations in a loop.
   SEXP env = PROTECT(r_new_environment(R_GlobalEnv, 2));
-  Rf_defineVar(syms_bracket, fns_bracket, env);
   Rf_defineVar(syms_x, x, env);
   Rf_defineVar(syms_i, info.index, env);
 
+  // Construct call with symbols, not values, for performance
+  SEXP call;
+  if (Rf_inherits(x, "integer64")) {
```

Comment:
Needs a comment about the reason for special-casing, and mention this should be removed once integer64 is fixed.

Reply:
Maybe add `is_integer64()` that would first check for the correct TYPEOF? Would be slightly faster for non-REALSXP types.